src/Symfony/Component/DependencyInjection/Compiler/ResolveNamedArgumentsPass.php

```diff
@@ -53,6 +53,10 @@ protected function processValue($value, $isRoot = false)
                     $parameters = $r->getParameters();
                 }
 
+                if (isset($key[0]) && '$' !== $key[0] && !class_exists($key)) {
+                    throw new InvalidArgumentException(sprintf('Invalid service "%s": did you forget to add the "$" prefix to argument %s?', $this->currentId, $key));
```

Comment:
`... to argument "%s"?` instead of `... to argument %s?`  We always try to wrap in quotes anything typed by the user to avoid potential confusion.